internal/cmd/schema-registry/command_schema_create.go

```diff
@@ -101,7 +108,35 @@ func (c *command) schemaCreate(cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
-	response, _, err := srClient.DefaultApi.Register(ctx, subject, srsdk.RegisterSchemaRequest{Schema: string(schema), SchemaType: schemaType, References: refs})
+	var metadata srsdk.Metadata
+	var ruleset srsdk.RuleSet
+	metadataPath, err := cmd.Flags().GetString("metadata")
+	if err != nil {
+		return err
+	}
+	rulesetPath, err := cmd.Flags().GetString("ruleset")
+	if err != nil {
+		return err
+	}
+	err = readMetadataAndRuleset(metadataPath, &metadata, rulesetPath, &ruleset)
+	if err != nil {
+		return err
+	}
+
+	normalize, err := cmd.Flags().GetBool("normalize")
+	if err != nil {
+		return err
+	}
+
+	request := srsdk.RegisterSchemaRequest{
+		Schema:     string(schema),
+		SchemaType: schemaType,
+		References: refs,
+		Metadata:   metadata,
+		RuleSet:    ruleset,
+	}
+	response, _, err := srClient.DefaultApi.Register(ctx, subject, request,
```

Comment:
Are there any error details in the second argument that we're ignoring here? Most of our APIs follow that pattern so it seems odd that we're choosing to ignore this.

Reply:
All of Confluent's other public APIs have a field called "details" in their error struct, which gets returned inside the HTTP response. Is that not the case here?

Reply:
https://github.com/confluentinc/schema-registry-sdk-go/blob/master/api_default.go#L598 
The second response is `*_nethttp.Response`

Reply:
Right, but for the majority of our APIs, this HTTP response contains a "details" field in the case of an error. Does your API return that field?

Reply:
No, doesn't look like so.